docs/tutorials/11_VarQTE.ipynb

```diff
@@ -0,0 +1,925 @@
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "mz_exp_val_re = np.array([ele[0][0] for ele in evolution_result_re.observables])\n",
+    "exact_mz_exp_val_re = sol.observables[0][0].real\n",
+    "times = evolution_result_re.times\n",
+    "pylab.plot(times, mz_exp_val_re, label= \"VarQRTE\")\n",
+    "pylab.plot(times, exact_mz_exp_val_re , label= \"Exact\",  linestyle='--')\n",
+    "pylab.xlabel(\"Time\")\n",
+    "pylab.ylabel(r\"$\\langle m_z \\rangle$\")\n",
+    "pylab.legend(loc=\"upper right\");"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "## Efficient Way to run VarQRTE\n",
+    "\n",
+    "Similarly, we can set `qpt` as `ReverseQGT()` and `gradient` as `ReverseEstimatorGradient()` to speed up the calculation."
+   ],
+   "metadata": {
+    "id": "korN4ZZv8j_8"
+   }
+  },
+  {
+   "cell_type": "code",
+   "source": [
+    "var_principle = RealMcLachlanPrinciple(qgt = ReverseQGT() , gradient = ReverseEstimatorGradient())\n",
+    "time = 10.0\n",
+    "evolution_problem = TimeEvolutionProblem(hamiltonian, time, aux_operators=aux_ops)\n",
+    "var_qrte = VarQRTE(ansatz, init_param_values,var_principle, Estimator())\n",
+    "evolution_result_re_eff = var_qrte.evolve(evolution_problem)"
+   ],
+   "metadata": {
+    "id": "gnBS19NQ85cn"
+   },
+   "execution_count": 24,
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/Users/ept/Desktop/repo-move/qiskit-algorithms/qiskit_algorithms/time_evolvers/variational/variational_principles/real_mc_lachlan_principle.py:161: UserWarning: A gradient instance with a setting for calculating real part of thegradient was provided. This variational principle requires theimaginary part. The setting to imaginary was changed automatically.\n",
+      "  warnings.warn(\n"
```

Comment:
Can we alter things so it does not emit a warning? It seems like something is not done right and for a tutorial I think it should be better